cli/azd/pkg/project/service_target_aks.go

```diff
@@ -0,0 +1,444 @@
+}
+
+type AksDeploymentOptions struct {
+	Name string `yaml:"name"`
+}
+
+type AksServiceOptions struct {
+	Name string `yaml:"name"`
+}
+
+type aksTarget struct {
+	config           *ServiceConfig
+	env              *environment.Environment
+	scope            *environment.TargetResource
+	containerService azcli.ContainerServiceClient
+	az               azcli.AzCli
+	docker           docker.Docker
+	kubectl          kubectl.KubectlCli
+	clock            clock.Clock
+}
+
+func (t *aksTarget) RequiredExternalTools() []tools.ExternalTool {
+	return []tools.ExternalTool{t.docker, t.kubectl}
+}
+
+func (t *aksTarget) Deploy(
+	ctx context.Context,
+	azdCtx *azdcontext.AzdContext,
+	path string,
+	progress chan<- string,
+) (ServiceDeploymentResult, error) {
+	// Login to AKS cluster
+	namespace := t.getK8sNamespace()
+	clusterName, has := t.env.Values[environment.AksClusterEnvVarName]
+	if !has {
+		return ServiceDeploymentResult{}, fmt.Errorf(
+			"could not determine AKS cluster, ensure %s is set as an output of your infrastructure",
+			environment.AksClusterEnvVarName,
+		)
+	}
+
+	// Login to container registry.
+	loginServer, has := t.env.Values[environment.ContainerRegistryEndpointEnvVarName]
+	if !has {
+		return ServiceDeploymentResult{}, fmt.Errorf(
+			"could not determine container registry endpoint, ensure %s is set as an output of your infrastructure",
+			environment.ContainerRegistryEndpointEnvVarName,
+		)
+	}
+
+	log.Printf("getting AKS credentials %s\n", clusterName)
+	progress <- "Getting AKS credentials"
+	credentials, err := t.containerService.GetAdminCredentials(ctx, t.scope.ResourceGroupName(), clusterName)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed retrieving cluster admin credentials, %w", err)
+	}
+
+	kubeConfigManager, err := kubectl.NewKubeConfigManager(t.kubectl)
+	if err != nil {
+		return ServiceDeploymentResult{}, err
+	}
+
+	kubeConfig, err := kubectl.ParseKubeConfig(ctx, credentials.Kubeconfigs[0].Value)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed parsing kube config: %w", err)
+	}
+
+	if err := kubeConfigManager.SaveKubeConfig(ctx, clusterName, kubeConfig); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed saving kube config: %w", err)
+	}
+
+	if err := kubeConfigManager.MergeConfigs(ctx, "config", "config", clusterName); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed merging kube configs: %w", err)
+	}
+
+	if _, err := t.kubectl.ConfigUseContext(ctx, clusterName, nil); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed using kube context '%s', %w", clusterName, err)
+	}
+
+	kubeFlags := kubectl.KubeCliFlags{
+		Namespace: namespace,
+		DryRun:    "client",
+		Output:    "yaml",
+	}
+
+	progress <- "Creating k8s namespace"
+	namespaceResult, err := t.kubectl.CreateNamespace(
+		ctx,
+		namespace,
+		&kubectl.KubeCliFlags{DryRun: "client", Output: "yaml"},
```

Comment:
Things like namespace, output type & dryrun are very common on many commands in kubectl.  Namespaces are created at the root and don't require a namespace flag.  For creating additional resources inside of a namespace you would pass the `namespace` flag.

The combination of `dryrun=client` && `output=yaml` provide us an easy way to create or update a resource.  The CLI basically generates a `yaml` k8s manifest of the specified resource on the `client` and returns the yaml without submitting it to the server.  We can then apply this yaml manifest using `kubectl apply -f - ` using `stdin` and if the resource does not exist it would be created otherwise updated.  

If you attempted to create a namespace or other resource using standard `kubectl create namespace` and it already exists it would fail.